File: examples/nodejs/send-phone.js
// Send an SMS through the paired phone — works anywhere in the world.
//   node send-phone.js

import { TextoryClient } from './client.js';

const apiKey = process.env.TEXTORY_API_KEY;
if (!apiKey) {
  console.error('Missing TEXTORY_API_KEY. export TEXTORY_API_KEY=sk_live_...');
  process.exit(1);
}

const tx = new TextoryClient(apiKey);

try {
  const result = await tx.sendPhone({
    recipients: [
      { phoneNumber: '+14155551234', name: 'Alice' },
    ],
    contents: 'Hi {{name}}, your verification code is 917043.',
  });

  console.log('✅ Sent:', JSON.stringify(result, null, 2));
} catch (err) {
  if (err.response) {
    console.error(`❌ ${err.response.status}:`, JSON.stringify(err.response.data, null, 2));
  } else {
    console.error('❌ Network error:', err.message);
  }
  process.exit(1);
}

File: examples/python/send_phone.py
"""Send an SMS via the paired phone (works globally)."""
import os
import sys

from textory_client import TextoryClient, TextoryError


def main() -> int:
    tx = TextoryClient()
    try:
        result = tx.send_phone(
            recipients=[{"phoneNumber": "+14155551234", "name": "Alice"}],
            contents="Hi {{name}}, your verification code is 917043.",
        )
    except TextoryError as err:
        print(f"❌ {err.status} {err.code}: {err.message}", file=sys.stderr)
        if err.details:
            print(f"  details: {err.details}", file=sys.stderr)
        return 1

    print("✅ Sent:")
    print(result)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
